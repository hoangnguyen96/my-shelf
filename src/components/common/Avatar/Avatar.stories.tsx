import type { Meta, StoryFn, StoryObj } from "@storybook/react";
import Avatar from ".";
import { RouterContext } from "next/dist/shared/lib/router-context.shared-runtime";
import { mockRouter } from "@app/mocks/storybook";
import { ChakraProvider } from "@chakra-ui/react";
import theme from "@app/themes";

const meta: Meta<typeof Avatar> = {
  component: Avatar,
  parameters: {
    nextjs: {
      appDirectory: true,
    },
  },
  decorators: [
    (Story: StoryFn) => (
      <ChakraProvider theme={theme}>
        <RouterContext.Provider value={mockRouter}>
          <Story />
        </RouterContext.Provider>
      </ChakraProvider>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Avatar>;

export const Default: Story = {
  args: {
    image: "https://i.ibb.co/SKHPQYq/avatar-default.webp",
  },
};
