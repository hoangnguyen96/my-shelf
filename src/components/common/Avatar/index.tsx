"use client";

import { memo } from "react";
import Image from "next/image";

interface AvatarProps {
  image: string;
  width?: number;
  height?: number;
  border?: string;
}

const Avatar = ({
  image,
  width = 100,
  height = 100,
  border = "none",
  ...rest
}: AvatarProps) => {
  return (
    <Image
      src={image || "https://i.ibb.co/SKHPQYq/avatar-default.webp"}
      width={width}
      height={height}
      alt="Avatar user"
      style={{
        maxWidth: "100%",
        borderRadius: "50%",
        border: border,
        height: "100%",
      }}
      priority={true}
      fetchPriority="high"
      {...rest}
    />
  );
};

export default memo(Avatar);
