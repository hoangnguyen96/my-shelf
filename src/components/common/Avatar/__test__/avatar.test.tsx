import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Avatar from "..";

describe("Avatar", () => {
  it("Should render correctly snapshot", () => {
    expect(
      render(
        <Avatar
          image="https://i.ibb.co/88X1WfZ/avatar-default.png"
          width={100}
          height={100}
        />
      )
    ).toMatchSnapshot();
  });

  it("Should render correctly snapshot when image empty", () => {
    expect(
      render(<Avatar image="" width={100} height={100} />)
    ).toMatchSnapshot();
  });
});
