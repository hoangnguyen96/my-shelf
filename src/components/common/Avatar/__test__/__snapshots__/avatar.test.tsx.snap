// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Should render correctly snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="Avatar user"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="100"
        src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F88X1WfZ%2Favatar-default.png&w=256&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F88X1WfZ%2Favatar-default.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F88X1WfZ%2Favatar-default.png&w=256&q=75 2x"
        style="color: transparent; max-width: 100%; max-height: 100%; border-radius: 50%;"
        width="100"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="Avatar user"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="100"
      src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F88X1WfZ%2Favatar-default.png&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F88X1WfZ%2Favatar-default.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F88X1WfZ%2Favatar-default.png&w=256&q=75 2x"
      style="color: transparent; max-width: 100%; max-height: 100%; border-radius: 50%;"
      width="100"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar Should render correctly snapshot when image empty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="Avatar user"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="100"
        src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=256&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=256&q=75 2x"
        style="color: transparent; max-width: 100%; max-height: 100%; border-radius: 50%;"
        width="100"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="Avatar user"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="100"
      src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=256&q=75 2x"
      style="color: transparent; max-width: 100%; max-height: 100%; border-radius: 50%;"
      width="100"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
